src/components/slop-fest-modal.jsx

```diff
@@ -0,0 +1,189 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useContext, useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+// import Select from "react-select";
+import { GET_USER_FESTS } from "../graphql";
+import { GET_USERS } from "../graphql/get-users";
+import { CREATE_FEST } from "../graphql/mutations/create-fest/create-fest";
+import { useModals } from "../store";
+import { CurrentUserContext } from "../store/current-user-context";
+import { Form, Modal } from "./index";
+
+export function SlopFestModal() {
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [users, setUsers] = useState([]);
+  const { currentUser } = useContext(CurrentUserContext);
+  const { closeModal } = useModals();
+  // const [attendees, setAttendees] = useState([]);
+
+  const { data, loading, error } = useQuery(GET_USERS, {
+    variables: {
+      where: {
+        id: {
+          not: {
+            equals: currentUser.id,
+          },
+        },
+      },
+    },
+  });
+  const [createFest, { loading: createLoading, error: createError }] =
+    useMutation(CREATE_FEST, { refetchQueries: [GET_USER_FESTS] });
+
+  const defaultValues = [{ username: currentUser.username }];
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { isValid, errors },
+    setValue,
+    getValues,
+  } = useForm({
+    defaultValues: {
+      name: "",
+      startDate: "",
+      endDate: "",
+      attendees: [],
+    },
+  });
+
+  useEffect(() => {
+    if (data && data.users) {
+      setUsers(data.users);
+    }
+  }, [data]);
+
+  const userOptions = users.map((user) => ({
+    username: user.username,
+  }));
+
+  const onSubmit = () => {
+    const { name, attendees } = getValues();
+    const startDateISO = startDate.toISOString().substring(0, 10);
+    const endDateISO = endDate.toISOString().substring(0, 10);
+    try {
+      createFest({
+        variables: {
+          data: {
+            name: name,
+            startDate: startDateISO,
+            endDate: endDateISO,
+            // attendees should include creator and other usernames in attendees field
+            attendees: {
+              connect: attendees,
+            },
+            creator: {
+              connect: { id: currentUser.id },
+            },
+          },
+        },
+      }).then(() => closeModal("create"));
+    } catch (error) {
+      console.log(`Error: ${error.message}`);
+    }
+  };
+
+  return (
+    <Modal title="OH WE FESTIN">
+      <Form
+        onSubmit={handleSubmit(onSubmit)}
+        className={"flex flex-col items-center p-6"}
+      >
+        <div className="flex flex-col">
+          <div className="flex-col">
+            <Form.TextInput
+              labelText="Name it!"
```

Comment:
Overall the modal works how it is suppose to but I did not notice some slight misalignments with the Figma design. These are not major and might be easy to fix. 

According to figma when the modal opens,  the name it field should have a placeholder of "Name it"
<img width="550" alt="Screenshot 2024-02-27 at 5 07 25 PM" src="https://github.com/jahorwitz/slopopedia/assets/109189846/1cf64a1d-58ff-41e7-88b9-875abaeeef84">

However, when I open it, it shows up with a placeholder saying "type here"
![Screenshot 2024-02-27 at 5 05 41 PM](https://github.com/jahorwitz/slopopedia/assets/109189846/daff2cd6-b7ef-4198-8bcb-0a86e3f125de)

Reply:
That's the way it is in the form component. I think that the form component might need a little refactoring which will affect a few components.

Reply:
Ahh I see! I think it could be added to another ticket. If we want to move this into final review. Up to you

Reply:
Let me talk with Josh and see what he thinks is best for this

Reply:
This has been resolved by Sam